### ISSUE TITLE
Jellying ImmutableSet

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#10403]
|<img alt="br1's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| br1 reported|
|-|-|
|Trac ID|trac#642|
|Type|enhancement|
|Created|2004-06-25 07:26:41Z|

<details><summary>Searchable metadata</summary>

```
trac-id__642 642
type__enhancement enhancement
reporter__br1 br1
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__pb pb
keywords__ 
time__1088148401000000 1088148401000000
changetime__1203243068000000 1203243068000000
version__ 
owner__ 
cc__warner cc__br1
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="br1's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|br1 commented|
|-|-|

```
#!html
<pre>
Currently, ImmutableSet is unjellyable, and if you try to send it through a pb 
connection, InsecureJelly is raised.  Is there a workaround, short of using a 
plain dictionary?
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="warner's avatar" src="https://avatars.githubusercontent.com/u/27146?s=50" width="50" height="50">](https://github.com/warner)|@warner commented|
|-|-|

```
#!html
<pre>
I've added support for Set and ImmutableSet to newpb, so the future is taken
care of :)

For oldpb, you'll probably have to hack jelly to have a 'if typ == sets.Set'
clause and create a new jelly sequence for it. It shouldn't be too hard,
basically cut-and-paste the code for lists.

We could get this fixed in oldpb, but I think newpb is close enough that it may
not be worth the effort. I could be wrong, however.

Or of course you can do an application-level workaround, but that will be
difficult to handle if the sets are buried deep inside some other data structure.

good luck,
 -Brian
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun commented|
|-|-|

PB should probably have support for sets added.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_5"></a>|@therve set status to `closed`|
|-|-|

(In [22570]) Merge set-jelly-2958

Author: therve
Reviewer: exarkun
Fixes [#2958](https://github.com/twisted/twisted/issues/2958)
Fixes [#10403](https://github.com/twisted/twisted/issues/10403)

Add support for set, frozenset, sets.Set and sets.ImmutableSet to jelly.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
